src/main/java/tests/CommentPostTests.java

```diff
@@ -46,18 +56,76 @@ public void setup() {
                 .webAppContextSetup(context)
                 .apply(springSecurity())
                 .build();
+        User user = new User("sarran", "theman");
+        if(userService.loadUserByUsername(user.getUsername()) == null) {
+            userService.registerNewUserAccount(user);
+        }
+    }
+
+    @Test
+    public void testAddCommentToValidSubmission() throws Exception {
+        SubmissionPost post = new SubmissionPost();
+        post.setTitle("This is my AMA");
+        post.setText("Test text.");
+        submissionPostRepository.save(post);
+        mvc
+                .perform(post("/posts/" + post.getTitle())
+                        .with(csrf())
+                        .with(user("sarran"))
+                        .param("text", "This is a test comment"))
```

Comment:
I'm not sure that will actually work? You can probably check that by querying the commentPostRepository and seeing if it's context was properly set